sdk/core/azure-core/src/base64.cpp

```diff
@@ -67,10 +67,11 @@ namespace Azure { namespace Core {
     BIO_flush(bio);
     BUF_MEM* bufferPtr;
     BIO_get_mem_ptr(bio, &bufferPtr);
+    std::string toReturn(bufferPtr->data, bufferPtr->length);
     BIO_set_close(bio, BIO_NOCLOSE);
```

Comment:
@vhvb1989, if we do all the copying before, would we still need `BIO_NOCLOSE` flag?
BTW, it also looks like it is caused by a bug in OpenSSL - https://github.com/openssl/openssl/issues/10448, but this fix makes us independent of whether the bug is there or not.

Reply:
I believed this implementation was migrated from storage, @Jinming-Hu  might have more experience using BIO.

We need unit test and change log for this.

I have set July milestone so this can be part of Core 1.1-beta1

Reply:
> We need unit test and change log for this.

@vhvb1989, we can add a `Bug Fixes` section to the changelog, and add an entry, after the following PR is merged in a couple of days.
https://github.com/Azure/azure-sdk-for-cpp/pull/2402

@jorgen, if at that time you are up for it, feel free to add an entry yourself. Great catch!
https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/CHANGELOG.md#release-history

Reply:
This issue and questions raised further highlights the priority of fixing the following issue: https://github.com/Azure/azure-sdk-for-cpp/issues/1338#issuecomment-797724760